pyroomacoustics/libroom_src/room.cpp

```diff
@@ -276,3 +276,422 @@ int Room::image_source_shoebox(
+			
+			Eigen::VectorXf hit_point_mic = mic_intersection(start, hit_point, mic_pos, mic_radius);
+			
+			//std::cout << "MICROPHONE hit_point : " << hit_point_mic[0] << " "<< hit_point_mic[1] << " "<< hit_point_mic[2] << std::endl;
+			distance = (start - hit_point_mic).norm();
+			
+			float travel_time_to_mic = travel_time;
+
+			update_travel_time(travel_time_to_mic, distance, sound_speed);
+			
+			if (travel_time_to_mic < time_thres){
+				//std::cout << "We append something" << std::endl;
+				append(travel_time_to_mic, energy, output);
+			}
+			
+		}
+		
+		
+		// Update the characteristics
+		
+		//std::cout << "\n\n-------\nprev travel time : " <<  travel_time <<std::endl;
+		//std::cout << "prev energy : " <<  energy <<std::endl;
+		distance = (start - hit_point).norm();
+		total_dist += distance;
+		update_travel_time(travel_time, distance, sound_speed);
+		update_energy_wall(energy, wall);
+		
+		//std::cout << "post travel time : " <<  travel_time <<std::endl;
+		//std::cout << "post energy : " <<  energy <<std::endl;
+		
+		// Check if we reach the time threshold for this ray
+		if (travel_time > time_thres){
+			break;
+		}
+		
+		// Let's simulate the scattered ray induced by the rebound on
+		// the wall
+		
+		if (scatter_coef > 0){
+			
+			float ray_scat_energy = compute_scat_energy(energy,
+														scatter_coef,
+														wall,
+														start,
+														hit_point,
+														mic_pos);
+														
+			// Shoot the scattered ray
+			scat_ray(wall,
+					 hit_point,
+					 mic_radius,
+					 ray_scat_energy,
+					 travel_time,
+					 time_thres,
+					 sound_speed,
+					 output);
+			
+			// The overall ray's energy gets decreased by the total
+			// amount of scattered energy
+			energy = energy*(1-scatter_coef);
+			
+		}
+				
+		// Now we just need to update the start and end positions for
+		// next hop
+		end = compute_reflected_end(start, hit_point, wall.normal, max_dist);
+		start = hit_point;		
+	}
+	
+	
+}
+
+std::vector<entry> Room::get_rir_entries(size_t nb_phis,
+								   size_t nb_thetas,
+								   Eigen::VectorXf source_pos,
+								   float mic_radius,
+								   float scatter_coef,
+								   float time_thres,
+								   float sound_speed){
+									   
+	/*This method simulates a certain number of rays and output the
+	 * vector of entries produced by the latter.
+	 * 
+	 * INPUT
+	 * - nb_phis is the number of different planar directions that will be used
+	 * - nb_thetas is the number of different elevation angles that will be used
+	 *   (NOTE : nb_phis*nb_thetas is the number of simulated raysUT
+	 * 
+	 * OUTPUT 
+	 * - a 2D vector where each entry is a tuple (time, energy)
+	 * reprensenting a ray (scattered or not) reaching the microphone*/
+	 
+	if ( (source_pos-mic_pos).norm() < mic_radius ){
+		std::cerr << "The source is inside the microphone ! " <<std::endl;
+		throw std::exception();
+	}
+	
+	std::vector<entry> output(10000);
```

Comment:
Here this should be just
```
std::vector<entry> output;
```
The vector will take of resizing when necessary.

Reply:
Done.